frontend/src/components/mine/Incidents/MineIncidents.js

```diff
@@ -70,8 +70,12 @@ const defaultProps = {
 };
 
 export class MineIncidents extends Component {
+  state = { isLoaded: false };
+
   componentDidMount() {
-    this.props.fetchMineIncidents(this.props.mineGuid);
+    this.props.fetchMineIncidents(this.props.mineGuid).then(() => {
+      this.setState({ isLoaded: true });
+    });
```

Comment:
Just wondering. Would it be better for this to be an array of requests passed to Promise.all() so that `isLoaded` could be set to `true` once all dependencies were fetched?

This question applies to files with similar logic as well.

Reply:
I was wondering this as well, through my testing it was never an issue, but local is much faster than other envs.